plot-geotiff-pixel-bbox.py

```diff
@@ -0,0 +1,63 @@
+#!/usr/bin/env python3
+###########################################################
+# Plots a GeoTiff with a pixel count bounding box overlay.
+# Use to determine pixel values for ground control points
+# when geo-referencing or creating/testing cutlines.
+#######################################################
+import argparse
+import logging
+import os
+import rasterio
+from rasterio import windows
+import matplotlib.pyplot as plt
+from matplotlib.patches import Rectangle
+
+logging.basicConfig(format='%(message)s', level=logging.INFO)
+logger = logging.getLogger()
+
+
+def run(args):
+    src_file = args.src
+    min_x = args.min_x
+    max_x = args.max_x
+    min_y = args.min_y
+    max_y = args.max_y
+
+    logging.info('Reading band 1 from ' + src_file)
+    raster_src = rasterio.open(src_file)
+
+    logging.info('Building overlay')
+    slice_raster = (slice(min_y, max_y), slice(min_x, max_x))
+    window_slice = windows.Window.from_slices(*slice_raster)
+
+    plt.imshow(raster_src.read(1))
+    plt.title(os.path.basename(src_file))
+    ax = plt.gca()
+    ax.add_patch(
+        Rectangle(
+            (window_slice.col_off, window_slice.row_off),
+            width=window_slice.width,
+            height=window_slice.height,
+            fill=True,
+            alpha=.5,
+            color="red"
+        )
+    )
+
+    plt.show()
+
+
+def main():
+    parser = argparse.ArgumentParser(description='Plots GeoTiff with a pixel value bbox overlay')
```

Comment:
I was a little confused about what the `-X/-x -y/-Y` arguments were for initially, could be useful to specify that they're to specify the corners of the box (even though that's probably totally obvious). It might also be useful to say that the coordinates are in _pixel_ coordinates, not geo-referenced coordinates.

Reply:
I've updated the script to take `lower left` and `upper right` pixel coordinates instead of min/max values. Thanks for the feedback on this as it was a bit confusing the original way.